skipchain/api_test.go

```diff
@@ -99,6 +99,73 @@ func TestClient_CreateRootInter(t *testing.T) {
 	}
 }
 
+func TestClient_GetUpdateChain(t *testing.T) {
+	// Create a small chain and test whether we can get from one element
+	// of the chain to the last element with a valid slice of SkipBlocks
+	local := onet.NewTCPTest(cothority.Suite)
+	defer waitPropagationFinished(t, local)
+	defer local.CloseAll()
+
+	conodes := 10
+	sbCount := conodes - 1
+	servers, roster, gs := local.MakeHELS(conodes, skipchainSID, cothority.Suite)
+	s := gs.(*Service)
+
+	c := newTestClient(local)
+
+	sbs := make([]*SkipBlock, sbCount)
+	var err error
+	sbs[0], err = makeGenesisRosterArgs(s, onet.NewRoster(roster.List[0:2]),
+		nil, VerificationNone, 3, 1)
+	log.ErrFatal(err)
+	log.Lvl1("Initialize skipchain.")
+	// init skipchain
+	for i := 1; i < sbCount; i++ {
+		newSB := NewSkipBlock()
+		newSB.Roster = onet.NewRoster(roster.List[i : i+2])
+		service := local.Services[servers[i].ServerIdentity.ID][skipchainSID].(*Service)
+		log.Lvl2("Doing skipblock", i, servers[i].ServerIdentity, newSB.Roster.List)
+		reply, err := service.StoreSkipBlock(&StoreSkipBlock{LatestID: sbs[i-1].Hash, NewBlock: newSB})
+		require.Nil(t, err)
+		require.NotNil(t, reply.Latest)
+		sbs[i] = reply.Latest
+	}
+
+	for i := 0; i < sbCount; i++ {
+		sbc, err := c.GetUpdateChain(sbs[i].Roster, sbs[i].Hash)
+		log.ErrFatal(err)
+		if !sbc.Update[0].Equal(sbs[i]) {
+			t.Fatal("First hash is not from our SkipBlock")
+		}
+		require.True(t, len(sbc.Update) > 0, "Empty update-chain")
+		if !sbc.Update[len(sbc.Update)-1].Equal(sbs[sbCount-1]) {
+			log.Lvl2(sbc.Update[len(sbc.Update)-1].Hash)
+			log.Lvl2(sbs[sbCount-1].Hash)
+			t.Fatal("Last Hash is not equal to last SkipBlock for", i)
+		}
+		for up, sb1 := range sbc.Update {
+			log.ErrFatal(sb1.VerifyForwardSignatures())
+			if up < len(sbc.Update)-1 {
+				sb2 := sbc.Update[up+1]
+				h1 := sb1.Height
+				h2 := sb2.Height
+				log.Lvl3("sbc1.Height=", sb1.Height)
+				log.Lvl3("sbc2.Height=", sb2.Height)
+
+				height := h1
+				if h2 < height {
+					height = h2
+				}
+				if !bytes.Equal(sb1.ForwardLink[height-1].Hash(),
```

Comment:
`Hash` is a `SkipBlockID` which has an `Equal`. So this could be:
```
require.True(t, sb1.ForwardLink[height-1].Hash().Equal(sb2.Hash),
"Forward-pointer of update", up, "is different from hash in", up+1 )
```
just in case a student happens to read our code to know how to use things ;)